### ISSUE TITLE
Adding Docker Submissions from Participants

### ISSUE THREAD

**Issue by bbearce** [codalab/codalab-competitions#2784]
Hi,

I have recently added docker submission code to my CodaLab instance. I want to do a pull request for both the django code and compute_worker however I know there is existing code to submit docker containers already and would like to compare my code to that first. I took the liberty of documenting my changes on a website that I would ask that someone take a look at. My thought it is, if this seems like the right way to implement such a feature then I'll go ahead and do a pull request for both the compute_worker and the django code itself as those were the main things I changed. 

May I ask someone to briefly scroll through this post? 

Link: https://bbearce.github.io/#notes/codalab/docker_submission/

I'd really like to check in changes to this project if my changes are good enough. Please let me know as this has been quite a project for me.

The working example is at: https://miccai2020.eastus.cloudapp.azure.com/

**Comment by Didayolo** [comment]
Hi,

Thank you for your work. If I understand correctly, the aim of this feature is to be able to allow participants to submit code submissions on their own docker images?

This seems very useful.

Also, I think it needs some security measures to block malicious docker images.

**Comment by bbearce** [comment]
Yes!!! Essentially ends the need for me to setup a specific environment and
make everyone use the same packages, we can allow them to setup a custom
environment inside their image. Security is a potential issue. Eric and I's
first thought was that if all they can submit is the image itself, we have
control over the volumes\mounts. That is the main issue I see. I'd like to
help on this. I have some prelim code. I might need the rest of the week to
work on something else, but will be returning to this.
